.github/workflows/maven-cd.yml

```
@@ -51,17 +51,22 @@ jobs:
       - name: Log should_release details
         id: log-should_release-details
         run: |
-          echo "## Job Filter Summary"                                  >> $GITHUB_STEP_SUMMARY
-          echo "\`\`\`"                                                 >> $GITHUB_STEP_SUMMARY
-          echo "================================"                 | tee -a $GITHUB_STEP_SUMMARY
-          echo "Release job filter details:"                      | tee -a $GITHUB_STEP_SUMMARY
-          echo "          VALIDATE_ONLY: ${VALIDATE_ONLY}"        | tee -a $GITHUB_STEP_SUMMARY
-          echo "              CI_STATUS: ${CI_STATUS}"            | tee -a $GITHUB_STEP_SUMMARY
-          echo "   INTERESTING_CATEGORY: ${INTERESTING_CATEGORY}" | tee -a $GITHUB_STEP_SUMMARY
-          echo "--------------------------------"                 | tee -a $GITHUB_STEP_SUMMARY
-          echo "=>       SHOULD_RELEASE: ${SHOULD_RELEASE}"       | tee -a $GITHUB_STEP_SUMMARY
-          echo "================================"                 | tee -a $GITHUB_STEP_SUMMARY
-          echo "\`\`\`"                                                 >> $GITHUB_STEP_SUMMARY
+          MD_CODE="\`\`\`"
+          GHA_SUMMARY=$(cat << END_OF_SUMMARY
+          ## Job Filter Summary
+          ${MD_CODE}
+          ================================
+          Release job filter details:
+                    VALIDATE_ONLY: ${VALIDATE_ONLY}
+                        CI_STATUS: ${CI_STATUS}
+             INTERESTING_CATEGORY: ${INTERESTING_CATEGORY}
+          --------------------------------
+          =>       SHOULD_RELEASE: ${SHOULD_RELEASE}
+          ================================
+          ${MD_CODE}
+          END_OF_SUMMARY
+          )
+          echo "${GHA_SUMMARY}" | tee -a $GITHUB_STEP_SUMMARY
```

More simply (untested),

```suggestion
          cat << END_OF_SUMMARY | tee -a $GITHUB_STEP_SUMMARY
          ## Job Filter Summary
          ${MD_CODE}
          ================================
          Release job filter details:
                    VALIDATE_ONLY: ${VALIDATE_ONLY}
                        CI_STATUS: ${CI_STATUS}
             INTERESTING_CATEGORY: ${INTERESTING_CATEGORY}
          --------------------------------
          =>       SHOULD_RELEASE: ${SHOULD_RELEASE}
          ================================
          ${MD_CODE}
          END_OF_SUMMARY
```

---

`cat << END_OF_SUMMARY | tee -a $GITHUB_STEP_SUMMARY` - this is what I started with, but for some reason it did not get logged to stdout - only step summary was filled. Did test it now within a bash shell and there it worked. Will commit it but will perform another GHA test run before giving green light.